akka-persistence-query/src/main/scala/akka/persistence/query/scaladsl/PagedPersistenceIdsQuery.scala

```diff
@@ -5,18 +5,17 @@
 package akka.persistence.query.scaladsl
 
 import akka.NotUsed
-import akka.persistence.state.scaladsl.DurableStateStore
 import akka.stream.scaladsl.Source
 
 /**
- * A plugin may optionally support this query by implementing this trait.
+ * A plugin (e.g. ReadJournal or DurableStateStore) may optionally support this query by implementing this trait.
  */
-trait CurrentDurableStatePersistenceIdsQuery[A] extends DurableStateStore[A] {
+trait PagedPersistenceIdsQuery {
```

Comment:
good name and good approach to share same interface

Reply:
I noticed now why this might not work well after all. Then it can't be used as:
```
DurableStateStoreRegistry(system).durableStateStoreFor[PagedPersistenceIdsQuery](pluginId)
```
because that `durableStateStoreFor` requires a `T <: scaladsl.DurableStateStore[_]`

similar with `readJournalFor` in `PersistenceQuery`.

It still works with the concrete impl but not with the interface like that.

Reply:
For Scala it could still work when written like this:

```
DurableStateStoreRegistry.get(system)
    .durableStateStoreFor[DurableStateUpdateStore[ScalaPbAny] with PagedPersistenceIdsQuery](storeSettings.storeIdentifier)
```

... but indeed I don't think that'll work for the javadsl.

Should we duplicate this into `PagedPersistenceIdsQuery` and `DurableStateStorePagedPersistenceIdsQuery` then, even if the latter is getting rather long?

Reply:
> Should we duplicate this into PagedPersistenceIdsQuery and DurableStateStorePagedPersistenceIdsQuery

yes, I see no other way. Would be strange if this didn't follow the same pattern as the other query traits.